engine/consensus/dkg/processor/engine.go

```diff
@@ -74,11 +79,18 @@ func New(
 	return &eng, nil
 }
 
-// SetEpoch changes the epoch counter of the engine.
-func (e *Engine) SetEpoch(epochCounter uint64) {
-	e.unit.Lock()
-	defer e.unit.Unlock()
-	e.epochCounter = epochCounter
+// SetDKGInstanceID changes the DKG instance identifier of the engine.
+func (e *Engine) SetDKGInstanceID(value string) {
+	e.dkgInstanceIDLock.Lock()
```

Comment:
I don't think this lock is necessary. 

What race condition could the lock prevent?